net/Socket.cpp

```diff
@@ -726,28 +726,14 @@ int SocketPoll::poll(int64_t timeoutMaxMicroS)
 
         if (itemsErased)
         {
-            const size_t itemsErasedPre = itemsErased;
-            itemsErased = 0; // correcting itemsErased
+            LOG_TRC("Scanning to removing " << itemsErased << " defunct sockets from "
+                    << _pollSockets.size() << " sockets");
```

Comment:
I think what I am looking for here is a revert of the complexity inherent in:

80246f7ac1140257d14162504d3550909d6ba681

Reverting that patch as step #1 in its entirety - and then adding whatever it is we wanted to do back again would make me significantly happier. Lets get the revert in - and then re-think that from scratch - I am suspicious of all the changes to the arithemetic there - the itemsAdded size comparisons etc. etc. if there is a big-fix or benefit from each change to the logic there - it should be split out, reviewed and ideally unit tested separately. We should bin the lock, atomic, etc. etc. there too.

Reply:
it was just logging I added. Not required.